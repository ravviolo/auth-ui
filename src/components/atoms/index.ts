export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
