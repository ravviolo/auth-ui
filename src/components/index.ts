export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
