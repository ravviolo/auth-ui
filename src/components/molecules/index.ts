export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
