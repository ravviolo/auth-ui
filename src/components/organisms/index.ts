export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
