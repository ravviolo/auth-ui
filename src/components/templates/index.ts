export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
