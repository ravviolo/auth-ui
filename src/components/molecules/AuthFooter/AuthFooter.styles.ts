import styled from 'styled-components';

import { Flex, TextButton } from 'components/atoms';

export const Footer = styled.footer`
  font-size: 0.8rem;
`;

export const FooterButton = styled(TextButton)`
  text-decoration: underline;
  text-transform: uppercase;
  font-size: inherit;
`;

export const FooterFlex = styled(Flex)`
  gap: 4px;
  justify-content: center;
  align-items: center;
`;
