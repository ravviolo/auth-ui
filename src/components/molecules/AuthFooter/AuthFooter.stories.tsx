import { ComponentMeta, ComponentStory } from '@storybook/react';

import { AuthFooter } from 'components/molecules';

export default {
  title: 'Molecules/AuthFooter',
  component: AuthFooter,
  args: {
    btnText: 'Link',
    text: 'Footer text.',
  },
} as ComponentMeta<typeof AuthFooter>;

export const Default: ComponentStory<typeof AuthFooter> = (args) => (
  <AuthFooter {...args} />
);
Default.storyName = 'AuthFooter';
