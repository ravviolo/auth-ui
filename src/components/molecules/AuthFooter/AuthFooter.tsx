import * as S from './AuthFooter.styles';

export interface Props {
  text: string;
  btnText: string;
  testId: string;
  onClick: () => void;
}

export const AuthFooter = ({ btnText, testId, onClick, text }: Props) => {
  return (
    <S.Footer data-testid={testId}>
      <S.FooterFlex direction="row" testId="flex-footer">
        <p>{text}</p>
        <S.FooterBtn
          label={btnText}
          testId="btn-footer"
          type="button"
          onClick={onClick}
        />
      </S.FooterFlex>
    </S.Footer>
  );
};
