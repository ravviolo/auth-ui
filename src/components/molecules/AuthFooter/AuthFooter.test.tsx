import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { AuthFooter } from 'components/molecules';

import { Props as AuthFooterProps } from './AuthFooter';

describe('AuthFooter', () => {
  const mockOnClick = jest.fn<AuthFooterProps['onClick'], []>();

  const props: AuthFooterProps = {
    btnText: 'Footer Button',
    text: 'Footer Text',
    testId: 'test-auth-footer',
    onClick: mockOnClick,
  };

  it('renders footer containing text and button', () => {
    render(<AuthFooter {...props} />);

    const authFooter = screen.getByTestId('test-auth-footer');
    expect(authFooter).toHaveTextContent('Footer Text');

    const footerBtn = within(authFooter).getByTestId('btn-footer');
    expect(footerBtn).toHaveTextContent('Footer Button');
  });

  it('displays button that handles click events', async () => {
    render(<AuthFooter {...props} />);

    const authFooter = screen.getByTestId('test-auth-footer');
    const footerBtn = within(authFooter).getByTestId('btn-footer');

    await userEvent.click(footerBtn);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const tree = create(<AuthFooter {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
