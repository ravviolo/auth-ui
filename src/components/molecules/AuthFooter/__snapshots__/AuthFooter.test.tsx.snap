// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFooter should match snapshot 1`] = `
<footer
  className="sc-kDDrLX kktpMP"
  data-testid="test-auth-footer"
>
  <div
    className="sc-jSMfEi iChCMC sc-crXcEl bUmuaF"
    data-testid="flex-footer"
    direction="row"
  >
    <p>
      Footer Text
    </p>
    <button
      className="sc-jqUVSM ebpvV sc-iqcoie bPYbbt"
      data-testid="btn-footer"
      onClick={[MockFunction]}
      type="button"
    >
      Footer Button
    </button>
  </div>
</footer>
`;
